lua/entities/terminator_nextbot_zambiepaper/shared.lua

```diff
@@ -0,0 +1,114 @@
+    return
+end
+
+function ENT:AdditionalInitialize()
+    BaseClass.AdditionalInitialize( self )
+    
+    self:SetSubMaterial( 0, "models/props_c17/paper01" )
+    self:SetColor( Color( 230, 220, 200 ) )
+    
+    self.HeightToStartTakingDamage = 100
+    self.FallDamagePerHeight = 0.3
+    self.DeathDropHeight = 500
+end
+
+ENT.MyClassTask = {
+    OnStart = function( self, data )
+        hook.Add( "PostEntityTakeDamage", "PaperZombie_BleedEffect_" .. self:GetCreationID(), function( target, dmg, took )
+            if not IsValid( self ) then 
+                hook.Remove( "PostEntityTakeDamage", "PaperZombie_BleedEffect_" .. self:GetCreationID() )
+                return 
+            end
+            if not took then return end
+            if dmg:GetAttacker() ~= self then return end
+            
+            self:ApplyBleedEffect( target )
+        end )
+    end,
+}
+
+function ENT:OnRemove()
+    hook.Remove( "PostEntityTakeDamage", "PaperZombie_BleedEffect_" .. self:GetCreationID() )
+end
+
+function ENT:ApplyBleedEffect( victim )
+    if not IsValid( victim ) then return end
+    if not victim:Health() or victim:Health() <= 0 then return end
+    
+    if victim.PaperZombie_Bleeding then return end
+    victim.PaperZombie_Bleeding = true
+    
+    local bleedDuration = self.BleedDuration
+    local bleedDamage = self.BleedDamage
+    local bleedTicks = self.BleedTicks
+    local tickDelay = bleedDuration / bleedTicks
+    
+    local effectdata = EffectData()
+    effectdata:SetOrigin( victim:GetPos() + Vector( 0, 0, 40 ) )
+    effectdata:SetColor( victim:GetBloodColor() )
+    effectdata:SetScale( 1 )
+    util.Effect( "bloodspray", effectdata )
+    
+    victim:EmitSound( "physics/flesh/flesh_squishy_impact_hard" .. math.random( 1, 4 ) .. ".wav", 70, math.random( 90, 110 ) )
+    
+    for i = 1, bleedTicks do
+        timer.Simple( tickDelay * i, function()
+            if not IsValid( victim ) or not victim:Health() or victim:Health() <= 0 then 
+                if IsValid( victim ) then
+                    victim.PaperZombie_Bleeding = nil
+                end
+                return 
+            end
+            
+            local dmg = DamageInfo()
+            dmg:SetDamage( bleedDamage )
+            dmg:SetAttacker( IsValid( self ) and self or game.GetWorld() )
+            dmg:SetInflictor( IsValid( self ) and self or game.GetWorld() )
+            dmg:SetDamageType( DMG_SLASH )
+            victim:TakeDamageInfo( dmg )
+            
+            if i % 2 == 0 then
+                local bloodeffect = EffectData()
+                bloodeffect:SetOrigin( victim:GetPos() + Vector( 0, 0, 40 ) )
+                bloodeffect:SetColor( victim:GetBloodColor() )
+                bloodeffect:SetScale( 0.5 )
+                util.Effect( "bloodspray", bloodeffect )
+            end
+            
+            if i == bleedTicks then
+                victim.PaperZombie_Bleeding = nil
+            end
+        end )
+    end

```

Comment:
This should be a proper timer.Create instead of a bunch of simple timers
Right now this code breaks and applies bleeding even after someone respawns.
You could fix that with a simple tweak, but converting the entire thing to a proper timer.Create, will fix that, and all the other issues that might crop up.

Also you should convert victim.PaperZombie_Bleeding to a timestamp
eg, victim.PaperZombie_BleedingUntil = CurTime() + 10
That way players won't get stuck as "bleeding" if the code errors and the bool is never cleaned up